.github/workflows/version.yml

```diff
@@ -36,91 +37,83 @@ jobs:
     needs: ["lint"]
     runs-on: ubuntu-latest
     outputs:
-      files: ${{ steps.changed_files.outputs.files }}
+      files: ${{ steps.calc_file_list.outputs.changed_files }}
+      sign_matrix: ${{ steps.calc_file_list.outputs.result }}
     steps:
     - name: Checkout the repository
       uses: actions/checkout@v4
     
     - name: Get changed files for push
       if: github.event_name == 'push'
       id: changed_files_push
-      uses: jitterbit/get-changed-files@v1
+      uses: masesgroup/retrieve-changed-files@v3.0.0
+      with:
+        format: 'json'
 
-    - name: Get changed files
-      id: changed_files
-      run: |
-        if [[ ${{ github.event_name }} == "push" ]]; then
-          echo "files=${{ steps.changed_files_push.outputs.all }}" >> $GITHUB_OUTPUT
-        else
-          echo "files=${{ github.event.inputs.files }}" >> $GITHUB_OUTPUT
-        fi
+    - name: Calculate files to sign and push
+      uses: actions/github-script@v7
+      id: calc_file_list
+      with:
+        script: |
+          const signed_files = "${{ env.SIGNED_FILES }}".split(' ')
+          if ("${{ github.event_name }}" === "push") {
+            changed_files = JSON.parse('${{ steps.changed_files_push.outputs.all }}')
+            // Sign all files in case this workflow changes.
+            if (changed_files.includes(".github/workflows/version.yml")) {
+              changed_files = [...new Set([...changed_files, ...signed_files])]
+            }
+            core.setOutput("changed_files", changed_files.join(' '))
```

Comment:
Do we really need to upload new files if they did not change?

Reply:
Ss we do no i know the intention of the change, this is fine.

Reply:
Yeah, both might make sense :man_shrugging: